jetcd-core/src/main/java/io/etcd/jetcd/WatchImpl.java

```diff
@@ -197,13 +208,33 @@ public void onNext(WatchResponse response) {
         }
 
         listener.onError(error);
-      } else {
-        if (response.getCompactRevision() != 0) {
-          listener.onError(newCompactedException(response.getCompactRevision()));
-        } else {
-          listener.onNext(new io.etcd.jetcd.watch.WatchResponse(response));
-          revision = response.getEvents(response.getEventsCount() - 1).getKv().getModRevision() + 1;
-        }
+      } else if (response.getCompactRevision() != 0) {
```

Comment:
do you think it still makes sense to propagate the empty list to `listener.onNext`, maybe client will want to react on it?

Reply:
@lburgazzoli